workers/unity/Packages/io.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -33,41 +42,29 @@ public class ViewDiff
 
         private readonly List<IComponentDiffStorage> componentStorageList = new List<IComponentDiffStorage>();
 
-        private readonly Dictionary<uint, Dictionary<uint, IComponentCommandDiffStorage>> componentIdToCommandIdToStorage =
-            new Dictionary<uint, Dictionary<uint, IComponentCommandDiffStorage>>();
+        private readonly Dictionary<uint, (int firstIndex, int count)> componentIdStorageRange =
+            new Dictionary<uint, (int firstIndex, int count)>();
+
+        private readonly Dictionary<(uint componentId, uint commandId), IComponentCommandDiffStorage> componentCommandToStorage =
+            new Dictionary<(uint componentId, uint commandId), IComponentCommandDiffStorage>();
 
         private readonly Dictionary<Type, ICommandDiffStorage> typeToCommandStorage =
             new Dictionary<Type, ICommandDiffStorage>();
 
-        private readonly List<ICommandDiffStorage> commandStorageList = new List<ICommandDiffStorage>();
+        private readonly List<IComponentCommandDiffStorage> commandStorageList = new List<IComponentCommandDiffStorage>();
```

Comment:
A small explanation on this in comments would go a long way :)

Reply:
A `IComponentCommandDiffStorage` implements `ICommandDiffStorage`
All instances in this list were the former, except for the world command storage that we manually added.
Instead, I now store the generated diff storages only, and the world storage is called manually. (This is only done for clearing afterwards)

Reply:
I'm referring more to the range dictionary :)